src/openms/source/ANALYSIS/QUANTITATION/PeptideAndProteinQuant.cpp

```diff
@@ -427,16 +455,30 @@ namespace OpenMS
           peptides.push_back(ab.first);
         }
       }
-
-      map<UInt64, DoubleList> abundances; // all peptide abundances by sample
+      // done selecting peptides for quantification
 
       // consider only the selected peptides for quantification:
-      for (auto & pep : peptides)
-      {       
-        for (auto & sa : prot_q.second.abundances[pep])
+      map<UInt64, DoubleList> abundances; // all peptide abundances by sample
+      map<UInt64, DoubleList> psm_counts; // all PSM counts by sample
+      for (const auto & pep : peptides) // for all selected peptides
+      { 
+        for (auto & sa : prot_q.second.abundances[pep]) // copy over abundances
         {
           abundances[sa.first].push_back(sa.second);
         }
+
+        for (auto & sa : prot_q.second.psm_counts[pep]) // copy over psm counts
+        {
+          psm_counts[sa.first].push_back(sa.second);
+        }
+      }
+
+      // summarize all peptides of this protein to get map: sample->psm_count
+      // note: we take all for spectral counting (not only the top n ones) and always sum up
```

Comment:
need to check again if this is the case

Reply:
ok the comment seems to be wrong. psm_count is only filled for the "selected peptides".
Hmm no exactly sure what to do here. If I would do pure spectral counting I would set top n to 0 (all).
Here quantification at feature-intensity level could be TOP3 so the question would be if still all should be used.

Reply:
I agree that you would most likely use "all" for spectral counting but I would leave options open if possible.